File: courses/23-javascript/lesson-01-js-intro/code/examples/example-01.js
// example-01.js - Hello World 与 console.log 调试

// 最简单的Hello World
console.log("Hello, JavaScript!");

// 输出不同类型的值
console.log("这是一个字符串");
console.log(42); // 这是一个数字
console.log(true); // 这是一个布尔值
console.log(null); // 这是空值
console.log(undefined); // 这是未定义

// 使用模板字符串输出多个值
const name = "小明";
const age = 18;
console.log(`你好，${name}！你今年${age}岁了。`);

// 输出对象和数组
console.log({ name: "张三", score: 95 });
console.log([1, 2, 3, 4, 5]);

// 调试技巧：给输出添加标签
console.log("调试信息:", "程序执行到这里了");
console.log("变量值:", { name, age });

File: courses/19-rust/lesson-01-rust-intro/code/examples/example-01.rs
// example-01.rs - Hello World 程序详解
// 这是 Rust 中最经典的入门程序

// main 函数是程序的入口点
fn main() {
    // println! 是一个宏（macro），不是函数
    // 宏在编译时展开，提供零成本抽象
    // ! 表示这是一个宏调用
    println!("Hello, world!");

    // 我们也可以打印更复杂的信息
    println!("欢迎来到 Rust 编程世界！");
    println!("Rust 让系统编程既安全又高效！");
}